### PR TITLE
fix: normalize oauth emails before matching users

### LINKED ISSUE #9487: [Bug]: Google OAuth fails when provider returns mixed-case email
### Error Message and Logs

With an existing local user `user@example.edu`, Google OAuth failed in two different ways depending on registration settings:

- With registration disabled: `These credentials do not match our records.`
- With registration enabled: `Failed to process callback from login provider.`

Added temporary logging showed:

```log
OAuth callback user received {"provider":"google","email":"UserName@example.edu","name":"Example User","id":"google-user-id"}
OAuth callback failed {"provider":"google","message":"SQLSTATE[23505]: Unique violation: 7 ERROR: duplicate key value violates unique constraint \"users_email_unique\" DETAIL: Key (email)=(user@example.edu) already exists.","exception":"Illuminate\\Database\\UniqueConstraintViolationException"}
```

### Steps to Reproduce

**Steps to Reproduce**

1. Create an existing Coolify user with a lowercase email like username@example.edu
2. Configure Google OAuth
3. Use a Google account that returns the same email with different casing, e.g. UserName@example.edu
4. Try to sign in with Google

**Expected Behavior**

Coolify should normalize the OAuth email before matching or creating users, so an existing user with the same email but different casing can log in successfully.

**Actual Behavior**

Coolify does a case-sensitive lookup with the raw OAuth email, does not find the existing user, and then attempts to create a duplicate user record. Because the model lowercases the email on save, insert fails with users_email_unique.

### Example Repository URL

_No response_

### Coolify Version

v4.0.0-beta.470

### Are you using Coolify Cloud?

No (self-hosted)

### Operating System and Version (self-hosted)

Ubuntu 24.04.4 LTS

### Additional Information

The fix is to normalize the OAuth email before lookup and before create.

### PR DESCRIPTION
**Pull request by tristanbob**
## Changes

- Normalize the OAuth email before looking up an existing user.
- Reuse the normalized email when creating a new OAuth user record.
- Add a regression test for the case where the provider returns a mixed-case email for an already-existing lowercase user.

(This is Tristan, a human, and have review this PR for accuracy)

## Issues

- Fixes #9487
- Possible duplicates: #5498, #6681

## Category

- [x] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Adding new one click service
- [ ] Fixing or updating existing one click service

## Preview

N/A. This is a backend auth bug fix with no UI change.

## AI Assistance

- [ ] AI was NOT used to create this PR
- [x] AI was used (please describe below)

**If AI was used:**

- Tools used: Codex / GPT-5
- How extensively: AI was used to investigate the bug, prepare the patch, and add a regression test. The final change was validated manually and tested locally before opening the PR.

## Testing

- Reproduced the bug locally with Google OAuth returning a mixed-case email for an existing user.
- Confirmed the old behavior failed to match the existing user and could fall through to a duplicate-email create path.
- Verified the fix logs in the existing user successfully when the provider returns a mixed-case email.
- Ran `php artisan test --compact tests/Feature/OauthControllerTest.php`
- Ran `vendor/bin/pint --dirty --format agent`

## Contributor Agreement

> [!IMPORTANT]
>
> - [x] I have read and understood the [contributor guidelines](https://github.com/coollabsio/coolify/blob/v4.x/CONTRIBUTING.md). If I have failed to follow any guideline, I understand that this PR may be closed without review.
> - [x] I have searched [existing issues](https://github.com/coollabsio/coolify/issues) and [pull requests](https://github.com/coollabsio/coolify/pulls) (including closed ones) to ensure this isn't a duplicate.
> - [x] I have tested all the changes thoroughly with a local development instance of Coolify and I am confident that they will work as expected when a maintainer tests them.

### CODE REVIEW

**Comment by Twest2** [comment]
Nice catch and fix @tristanbob for the mixed-case lookup issue. There's one huge edge case that needs to be looked at though, if a provider returns no email, `(string) $oauthUser->email` will turn that into `''`, so the callback can continue with an empty email instead of failing.

That means we could end up querying `whereEmail('')`, and if registration is enabled, potentially creating an OAuth user with an empty email. 

Can you add an explicit non-empty email check before normalizing plus a small regression test for `null` / blank provider emails?

TLDR: One huge edge case: casting `$oauthUser->email` to string means a missing email becomes `''` so the OAuth flow can continue with an empty email rather than failing.

My suggested fix is...

```php
$oauthUser = get_socialite_provider($provider)->user();

$email = trim((string) $oauthUser->email);

if ($email === '') {
    abort(403, 'OAuth provider did not return an email address');
}

$email = strtolower($email);
$user = User::whereEmail($email)->first();

if (! $user) {
    $settings = instanceSettings();
    if (! $settings->is_registration_enabled) {
        abort(403, 'Registration is disabled');
    }

    $user = User::create([
        'name' => $oauthUser->name,
        'email' => $email,
    ]);
}

File: app/Http/Controllers/OauthController.php [hunk 1]
```diff
@@ -19,7 +19,8 @@ public function callback(string $provider)
     {
         try {
             $oauthUser = get_socialite_provider($provider)->user();
-            $user = User::whereEmail($oauthUser->email)->first();
+            $email = strtolower(trim((string) $oauthUser->email));
```
Comment (Twest2) [inline app/Http/Controllers/OauthController.php]: On null inputs, `(string) $oauthUser->email)` turns into `''` which allows empty emails to be processed.

Comment (tristanbob) [reply app/Http/Controllers/OauthController.php]: Thanks, this was a valid edge case.

Changes in this revision:

- fail fast when the OAuth provider returns an empty email address
- added regression coverage for both null and whitespace-only provider emails
- anonymized the test fixtures so the PR now uses generic data:
    - username@example.edu
    - UserName@example.edu
    - Example User

Verification:

- php artisan test --compact tests/Feature/OauthControllerTest.php
- vendor/bin/pint --dirty --format agent

I rewrote the branch history so the current PR branch only contains the sanitized test data.

**Comment by andrasbacsai** [comment]
Thank you for the PR! 💜
